tests/bsim/bluetooth/host/privacy/peripheral/src/tester.c

```diff
@@ -11,53 +11,72 @@
 #include <zephyr/bluetooth/bluetooth.h>
 
 #define EXPECTED_NUM_ROTATIONS 5
+#define MANUFACTURER_DATA_LEN  2
 
-static bt_addr_le_t old_addr;
-static int64_t old_time;
-static int rpa_rotations;
+struct adv_set_data_t {
+	bt_addr_le_t	old_addr;
+	int64_t	old_time;
+	int	rpa_rotations;
+};
+
+static  char	manufacturer_data[MANUFACTURER_DATA_LEN];
+static	struct adv_set_data_t adv_set_data[CONFIG_BT_EXT_ADV_MAX_ADV_SET];
+
+static bool data_cb(struct bt_data *data, void *user_data)
+{
+	switch (data->type) {
+	case BT_DATA_MANUFACTURER_DATA:
+		memcpy(manufacturer_data, data->data, MANUFACTURER_DATA_LEN-1);
+		manufacturer_data[MANUFACTURER_DATA_LEN] = '\0';
```

Comment:
I'm confused about this: You define it as a `char` and then NULL terminates it here, which is OK I guess (NULL termination only makes sense if you want to treat it as a string, which i don't see you doing, as you store an index in `manufacturer_data[0]`. 

Suggest to either make it a string (and use it properly as a string), or define it as `uint8_t` without adding string values to it.

Reply:
changed to uint8_t.